execution/src/execution.cc

```diff
@@ -126,7 +126,11 @@ namespace leatherman { namespace execution {
             }
         }
 
-        file = which(file, directories, expand);
+        try{
+            file = which(file, directories, expand);
+        } catch (const std::invalid_argument &ex) {
+            throw ex;
+        }
```

Comment:
I don't think we need this if we're already throwing the exception below, but I might be wrong.

Reply:
Intention is to propagate exception to the caller

Reply:
why though? seems redundant since you're just throwing back the same exception. but ok